sljit_src/sljitExecAllocator.c

```diff
@@ -115,19 +115,14 @@ static SLJIT_INLINE int get_map_jit_flag()
 	sljit_sw page_size;
 	void *ptr;
 	struct utsname name;
-	int os_kernel_major;
 	static int map_jit_flag = -1;
 
 	if (map_jit_flag < 0) {
 		map_jit_flag = 0;
 		uname(&name);
-		os_kernel_major = atoi(name.release);
 
-		/* this is likely to cause problems as seen in PCRE BUG2334 */
-		if (os_kernel_major > 19)
-			map_jit_flag = MAP_JIT;
-		else if (os_kernel_major >= 18) {
-			/* Kernel version for 10.14.0 (Mojave) or later */
+		/* Kernel version for 10.14.0 (Mojave) or later */
+		if (atoi(name.release) >= 18) {
```

Comment:
alternatively, if MAP_JIT is considered safe could keep the original logic but most likely only special casing Mojave for doing the test